### PR TITLE
feat: retain fragments while switching

### LINKED ISSUE #616: retain fragments while switching inbetween
**Describe the feature you'd like**
Fragment should not be lost while switching in between

**Would you like to work on the issue?**

- [x] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by simarsingh24**
fixes #616

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/MainActivity.kt [hunk 1]
```diff
@@ -106,9 +107,18 @@ class MainActivity : AppCompatActivity() {
         return super.onPrepareOptionsMenu(menu)
     }
 
+    private fun checkAndLoadFragment(fragment: Fragment) {
+        val frag = supportFragmentManager.findFragmentByTag(fragment::class.java.name)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/MainActivity.kt]: Rename to `savedFragment`

File: app/src/main/java/org/fossasia/openevent/general/MainActivity.kt [hunk 2]
```diff
@@ -106,9 +107,18 @@ class MainActivity : AppCompatActivity() {
...
+        if (frag != null) {
+            loadFragment(frag)
+            Timber.d("Loading fragment from stack...")
+        } else loadFragment(fragment)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/MainActivity.kt]: Wrap in curly braces

File: app/src/main/java/org/fossasia/openevent/general/MainActivity.kt [hunk 3]
```diff
@@ -106,9 +107,18 @@ class MainActivity : AppCompatActivity() {
         return super.onPrepareOptionsMenu(menu)
     }
 
+    private fun checkAndLoadFragment(fragment: Fragment) {
+        val frag = supportFragmentManager.findFragmentByTag(fragment::class.java.name)
+        if (frag != null) {
+            loadFragment(frag)
+            Timber.d("Loading fragment from stack...")
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/MainActivity.kt]: Add class name in log to show which fragment is being loaded

File: app/src/main/java/org/fossasia/openevent/general/order/OrderApi.kt [hunk 4]
```diff
@@ -2,11 +2,7 @@ package org.fossasia.openevent.general.order
 
 import io.reactivex.Single
 import org.fossasia.openevent.general.attendees.Attendee
-import retrofit2.http.Body
-import retrofit2.http.PATCH
-import retrofit2.http.GET
-import retrofit2.http.POST
-import retrofit2.http.Path
+import retrofit2.http.*
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/order/OrderApi.kt]: Irrelevant change

**Comment by simarsingh24** [comment]
@nikit19 @iamareebjamal please review and merge

**Pull request merged**